cdn.go

```diff
@@ -0,0 +1,241 @@
+// This file is part of gobizfly
+//
+// Copyright (C) 2020  BizFly Cloud
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>
+
+package gobizfly
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+const (
+	domainPath = "/domains"
+)
+
+type cdnService struct {
+	client *Client
+}
+
+var _ CDNService = (*cdnService)(nil)
+
+type CDNService interface {
+	List(ctx context.Context, opts *ListOptions) (*DomainsResp, error)
+	Get(ctx context.Context, domainID string) (*ExtendedDomain, error)
+	Create(ctx context.Context, cdrq *CreateDomainReq) (*CreateDomainResp, error)
+	Update(ctx context.Context, domainID string, udrq *UpdateDomainReq) (*UpdateDomainResp, error)
+	Delete(ctx context.Context, domainID string) error
+	DeleteCache(ctx context.Context, domainID string, files *Files) error
+}
+
+type User struct {
+	ID        int    `json:"id"`
+	Email     string `json:"email"`
+	TenantID  string `json:"tenant_id"`
+	CreatedAt string `json:"created_at"`
+	Name      string `json:"name"`
+	StartedAt string `json:"started_at"`
+	Status    string `json:"status"`
+}
+
+type UsageUser struct {
+	User  User `json:"user"`
+	Usage int  `json:"usage"`
+}
+
+type Domain struct {
+	ID             int    `json:"id"`
+	User           int    `json:"user"`
+	Certificate    int    `json:"certificate"`
+	CName          string `json:"cname"`
+	UpstreamAddrs  string `json:"upstream_addrs"`
+	Slug           string `json:"slug"`
+	PageSpeed      int    `json:"pagespeed"`
+	UpstreamProto  string `json:"upstream_proto"`
+	DDOSProtection int    `json:"ddos_protection"`
+	Status         string `json:"status"`
+	CreatedAt      string `json:"created_at"`
+	DomainID       string `json:"domain_id"`
+	Domain         string `json:"domain"`
+	Type           string `json:"type"`
+}
+
+type DomainsResp struct {
+	Domains []Domain `json:"results"`
+	Next    string   `json:"next"`
+	Prev    string   `json:"prev"`
+	Pages   int      `json:"pages"`
```

Comment:
Because these responses are paginate, Do you mind change the function to work with multiple pages?